extra_foam/gui/image_tool/reference_view.py

```diff
@@ -65,6 +77,21 @@ def updateF(self, data, auto_update):
             # is deferred.
             self._reference.setImage(data.image.reference)
 
+        if self._is_recording:
+            # Update cached image and count
+            image = self._corrected.image.astype(np.float32)
```

Comment:
Why is it necessary to explicitly convert to `np.float32`?

Reply:
Good point. This is indeed an expensive op. Do it only if necessary. Most likely the image here will be the assembled one and that should be of `dtype` `np.float32`

Reply:
Whoops, it's because I added tests of `int32` arrays. I forgot that the assembled images results to `np.float32`, so conversion shouldn't matter.